auraed/src/runtime/cell_service/executables/executable.rs

```diff
@@ -47,168 +39,89 @@ impl Executable {
     pub fn new<T: Into<ExecutableSpec>>(spec: T) -> Self {
         let ExecutableSpec { name, description, command } = spec.into();
         let state = ExecutableState::Init { command };
-        let inner = Arc::new(Mutex::new(ExecutableInner {
-            name: name.clone(),
-            description,
-            state,
-            stdout: LogChannel::new(format!("{name}::stdout")),
-            stderr: LogChannel::new(format!("{name}::stderr")),
-        }));
-        Self { inner: inner.clone(), log_thread: Some(spawn_log_thread(inner)) }
-    }
-
-    pub fn name(&self) -> io::Result<ExecutableName> {
-        let inner = self
-            .inner
-            .lock()
-            .map_err(|e| io::Error::new(io::ErrorKind::Other, e.to_string()))?;
-        Ok(inner.name.clone())
-    }
-
-    pub fn description(&self) -> io::Result<String> {
-        let inner = self
-            .inner
-            .lock()
-            .map_err(|e| io::Error::new(io::ErrorKind::Other, e.to_string()))?;
-        Ok(inner.description.clone())
+        Self { name, description, state }
     }
 
     /// Starts the underlying process.
     /// Does nothing if [Executable] has previously been started.
     pub fn start(&mut self) -> io::Result<()> {
-        let mut inner = self
-            .inner
-            .lock()
-            .map_err(|e| io::Error::new(io::ErrorKind::Other, e.to_string()))?;
-        let ExecutableState::Init { command } = &mut inner.state else {
+        let ExecutableState::Init { command } = &mut self.state else {
             return Ok(());
         };
 
-        let process = exec(command)?;
+        let mut child = command
+            .current_dir("/")
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .spawn()?;
+
+        let stdout = child.stdout.take().expect("stdout");
+        let log_channel = LogChannel::new(format!("{}::stdout", self.name));
+        let span = info_span!("running process", name = ?self.name);
```

Comment:
I think this should be inside the tokio spawn thread?

Reply:
ah gotcha. you're right, of course.  i keep forgetting the extra "entered" call that's necessary to enable the span.